Processing/BRCA_Deep_Dive.Rmd

```diff
@@ -2175,22 +1841,849 @@ ggplot() +
+  rename_with(~paste0(., "_csf"),-c("date", "id","name")) %>%
+  dplyr::mutate(id = as.factor(id))
+  #dplyr::mutate(ym = ym(paste0(year(date), "-", month(date)))) %>%
+  #dplyr::select(-date) %>%
+  #dplyr::group_by(ym,name,CF_csf,id) %>% 
+  #dplyr::summarize_all(sum) 
+
+csh <-left_join(clim_source_hist, 
+                source_hist_pts %>% 
+                  st_drop_geometry() %>% 
+                  dplyr::select(x,y,id), by = c("x","y")) %>% 
+  dplyr::filter(year(date) > 2005) %>% 
+  ungroup() %>% 
+  dplyr::select(-c(x,y)) %>% 
+  rename_with(~paste0(., "_csh"),-c("date", "id","name")) %>%
+  dplyr::mutate(id = as.factor(id)) #%>%
+ # dplyr::mutate(ym = ym(paste0(year(date), "-", month(date)))) %>%
+  #dplyr::select(-date) %>%
+  #dplyr::group_by(ym,name,CF_csh,id) %>% 
+  #dplyr::summarize_all(sum)
+
+bias <- left_join(csf,csh, by = c("date", "id","name")) %>% dplyr::arrange(ppt_mm_csf)
+
+bias_ww <- bias %>% 
+  dplyr::filter(CF_csf == "Warm Wet") %>%
+  dplyr::arrange(ppt_mm_csf) %>%
+  dplyr::mutate(ppt_csh_sort = sort(ppt_mm_csh))
+
+bias_hd <- bias %>% 
+  dplyr::filter(CF_csf == "Hot Dry") %>%
+  dplyr::arrange(ppt_mm_csf) %>%
+  dplyr::mutate(ppt_csh_sort = sort(ppt_mm_csh))
+
+ggplot() +
+  geom_point(data = bias_hd,
+             aes(x = ppt_csh_sort, y = ppt_mm_csf, color = "Hot Dry")) +
+  stat_smooth(data = bias_hd, 
+              aes(x = ppt_csh_sort, y = ppt_mm_csf, color = "Hot Dry"),
+              method = "lm", se = FALSE, formula = y~x, geom = "line", alpha = 0.5) +
+  geom_point(data = bias_ww,
+             aes(x = ppt_csh_sort, y = ppt_mm_csf, color = "Warm Wet")) +
+  stat_smooth(data = bias_ww, 
+              aes(x = ppt_csh_sort, y = ppt_mm_csf, color = "Warm Wet"),
+              method = "lm", se = FALSE, formula = y~x, geom = "line", alpha = 0.5) +
+  theme_bw() + 
+  scale_color_manual("", values = c("tomato", "dodgerblue")) +
+  geom_abline() +
+  facet_wrap(~factor(id)) +
+  labs(x = "GridMET Precip (mm)", y = "MACA Precip (mm)")
+  
+
+
+ # Bias correction using daily mean of all grid cells
+
+csf <- clim_source_fut %>%
+  dplyr::filter(year(date) < 2023) %>% 
+  dplyr::ungroup() %>%
+  dplyr::select(-c(x,y)) %>%
+  group_by(date, name, CF) %>%
+  dplyr::summarize_all(mean) %>%
+  rename_with(~paste0(., "_csf"),-c("date","name")) 
+
+csh <- clim_source_hist %>%
+  dplyr::filter(year(date) < 2023) %>% 
+  dplyr::ungroup() %>%
+  dplyr::select(-c(x,y)) %>%
+  group_by(date, name, CF) %>%
+  dplyr::summarize_all(mean) %>%
+  rename_with(~paste0(., "_csh"),-c("date","name")) 
+
+
+bias <- left_join(csf,csh, by = c("date","name")) %>% ungroup()
+bias_ww <- bias %>% 
+  dplyr::filter(CF_csf == "Warm Wet") %>%
+  dplyr::arrange(ppt_mm_csf) %>%
+  dplyr::mutate(ppt_csh_sort = sort(ppt_mm_csh))
+bias_hd <- bias %>% 
+  dplyr::filter(CF_csf == "Hot Dry") %>%
+  dplyr::arrange(ppt_mm_csf) %>%
+  dplyr::mutate(ppt_csf_sort = sort(ppt_mm_csf),
+                ppt_csh_sort = sort(ppt_mm_csh))
+
+
+ggplot() +
```

Comment:
Decided to comment out this section for now. It requires you to get climate data and run WBM for an overlapping date range -- which is not currently done in the code. Keeping it because Dave asked a question about why there's a gap in temperature between GridMET and MACA data. I could foresee having to address this and we might go back to discussions about bias correction. Currently, we don't do any. @ccmothes, in case you weren't part of the conversations:  Katie and I talked to Stephanie about using the overlap period for which GridMET and MACA are both available (2006-2024) and using that to develop a linear regression to bias correct the projections. HOWEVER, the timing of projected temp/precip (MACA) differs from GridMET because projections are statistical probabilities of future outcomes and trends, not a projection that a specific year will have a specific temp. So, in order to develop a regression, you first need to rank or sort the data and then regress that. However, you would have to  either a) do that for each grid cell, and the regression differs for each grid cell, or b) do it on the already summarized data. And we kinda just dropped the effort there. We might need to pick this back up so I'm keeping it commented out for now. Obviously there are published methods that we'll need to explore in order to move forward with this.